### ISSUE TITLE
Fail to create resource

### ISSUE THREAD

**Issue by thkdog** [emqx/emqx#2741]
#### Environment

- OS: CentOS7
- Erlang/OTP: R21/10.3.5.1
- EMQ: 3.2.0

#### Description

*A description of the issue*

I'd like to add web hook to notify a web service, but it says 'Start Resource resource:e81ce62a Failed' with a popup.

I have checked the logs in `/var/log/emqx` as following:

```
2019-07-31 17:01:56.119 [error] [RuleEngineAPI] init_resource_failure: {{emqx_web_hook_actions,
                                         on_resource_create},
2019-07-31 17:02:37.901 [error] cluster_call error found, ResL: [{{init_resource_failure,'emqx@127.0.0.1'},
                                  {{emqx_web_hook_actions,on_resource_create},
2019-07-31 17:02:37.901 [error] [RuleEngineAPI] test_resource_failure: [{{init_resource_failure,
                                           on_resource_create},
```

**Comment by 404div** [comment]
encountered this problem too.
any suggestiones, tks?

**Comment by thkdog** [comment]
The logs are lack of details, I can't tell anything from this problem...

**Comment by hashforu** [comment]
i'm not sure  if it's my setting error or bug that occured this.
waiting for offical reply or fix .

**Comment by terry-xiaoyu** [comment]
It is probably caused by the webhook URL can not be reached. 

Test the connection before creating the webhook resource. If the testing passed, the creation should be successful. Try emqx 3.2.2 as lots of issues have been fixed.

**Comment by hashforu** [comment]
created successfuly,thx.

**Comment by saraAlizadeh** [comment]
> 
> 
> created successfuly,thx.

how did you solve the issue? trying new version?

**Comment by seguidor777** [comment]
Same issue for me on version 4.2.5

`2021-04-14 00:30:22.805 [error] [RuleEngineAPI] init_resource_failure: {{emqx_web_hook_actions,
                                         on_resource_create},`
                                         
The endpoint is reachable from the cluster. I'm testing with this one https://emqx-test.free.beeceptor.com

**Comment by HJianBo** [comment]
Hi @seguidor777 could you give me more details log?

**Comment by seguidor777** [comment]
Sure
These logs appear when I test the connection from the UI.
```
(emqx@emqx-0.emqx-headless.tenant.svc.cluster.local)1> 2021-04-14 05:22:01.283 [error] cluster_call error found, ResL: [{{init_resource_failure,
                                   'emqx@emqx-0.emqx-headless.tenant.svc.cluster.local'},
                                  {{emqx_web_hook_actions,on_resource_create},
                                   {{badmatch,
                                     #{host =>
                                        <<"emqx-test.free.beeceptor.com">>,
                                       path => <<>>,scheme => <<"https">>}},
                                    [{emqx_web_hook_actions,pool_opts,1,
                                      [{file,"emqx_web_hook_actions.erl"},
                                       {line,296}]},
                                     {emqx_web_hook_actions,
                                      on_resource_create,2,
                                      [{file,"emqx_web_hook_actions.erl"},
                                       {line,165}]},
                                     {emqx_rule_engine,
                                      '-init_resource/4-fun-0-',4,
                                      [{file,"emqx_rule_engine.erl"},
                                       {line,437}]},
                                     {emqx_rule_engine,init_resource,4,
                                      [{file,"emqx_rule_engine.erl"},
                                       {line,437}]},
                                     {rpc,'-handle_call_call/6-fun-0-',5,
                                      [{file,"rpc.erl"},{line,197}]}]}}}]
(emqx@emqx-0.emqx-headless.tenant.svc.cluster.local)1> 2021-04-14 05:22:01.283 [error] [RuleEngineAPI] test_resource_failure: [{{init_resource_failure,
                                          'emqx@emqx-0.emqx-headless.tenant.svc.cluster.local'},
                                         {{emqx_web_hook_actions,
                                           on_resource_create},
                                          {{badmatch,
                                            #{host =>
                                               <<"emqx-test.free.beeceptor.com">>,
                                              path => <<>>,
                                              scheme => <<"https">>}},
                                           [{emqx_web_hook_actions,pool_opts,
                                             1,
                                             [{file,
                                               "emqx_web_hook_actions.erl"},
                                              {line,296}]},
                                            {emqx_web_hook_actions,
                                             on_resource_create,2,
                                             [{file,
                                               "emqx_web_hook_actions.erl"},
                                              {line,165}]},
                                            {emqx_rule_engine,
                                             '-init_resource/4-fun-0-',4,
                                             [{file,"emqx_rule_engine.erl"},
                                              {line,437}]},
                                            {emqx_rule_engine,init_resource,
                                             4,
                                             [{file,"emqx_rule_engine.erl"},
                                              {line,437}]},
                                            {rpc,
                                             '-handle_call_call/6-fun-0-',5,
                                             [{file,"rpc.erl"},
                                              {line,197}]}]}}}]
```

**Comment by HJianBo** [comment]
What about manually adding the port number and trying again? e.g: `https://emqx-test.free.beeceptor.com:80`

**Comment by seguidor777** [comment]
Thanks @HJianBo, it works now, I was just missing to specify the port

**Comment by HJianBo** [comment]
That's a bad design actually. We had fixed it in the v4.2.9 @seguidor777
